### DISCUSSION TITLE
conditional event checking if multiball is enabled is not working

### DISCUSSION THREAD

**Discussion by ohmbrew** [missionpinball/missionpinball#11] [Q&A] [answered]
Hello,

I have a multiball declared as:

multiballs:
  pt_multiball:
    debug: true
    ball_count: 3
    ball_count_type: total
    shoot_again: 5s
    start_events: start_multiball
    ball_locks: bd_pongcup1, bd_pongcup2

In the same mode, I want to immediately let the ball pass through the lock if we are in multiball. I thought this would work:

event_player:
  shot_pongcup1_hit{device.multiballs.pt_multiball.enabled}: release_cup1

But the ball is always being released - even before multiball is active. Am I misunderstanding what "enabled" means? This works a little better, but I'm not sure if it's reliable:

event_player:
  shot_pongcup1_hit{device.multiballs.pt_multiball.balls_live_target > 0}: release_cup1

**Comment by borgdog** [comment, marked as answer]
enabled is different from started, basically you are ready to start MB but need to hit a shot or whatever to do so.  I don't see a monitorable property for started for multiballs however.

looking at my setup, I have my multiball disable the lock when started, so any ball that enters the lock is automatically ejected until I reenable the lock. not sure if that would work in your setup having seen your rotating cup setup (which is very cool btw). 

another idea depending on your setup, my multiballs are all controlled by a state machine, and I can monitor the 'state' to see if any of mine are running.

**Reply by ohmbrew** [reply]
Well shit...I missed that one staring me right in the face only a few lines above this one - it seems checking the state of my multiball state machine is the way to go. Go figure.

Seems to be letting the balls pass through on multiball, and locking them when not. I have other issues with the mech getting confused when one ball is locked and the player drains - but at least I can move on from this issue. Thanks!!

Greg
